### PR TITLE
Bug fix

### LINKED ISSUE #255: [BUG] - Overflow of home page
### Is there any existing issue for this?

- [X] I have searched the existing issues

### Description

In continuation of issue #161 : It seems that the overflow of landing page was fixed in the PR created in response to the issue but in case of sub-sections issue remain un-resolved causing screen to scroll even when there is no extra content so It would be great if this issue is resolved for remaining sections as well !

### Screenshots

https://user-images.githubusercontent.com/115401171/221360315-21c8d8ac-d2c4-453f-b382-cb774b5caadd.mp4

### Any additional information?

_No response_

### What browser are you seeing the problem on?

Chrome, Microsoft Edge

### PR DESCRIPTION
**Pull request by ujjawaltyagii**
## Fixes Issue
- fixed overflow of homepage when there is : less content and no content
- made footer go scroll at end if there is more scrollable content
- responsiveness : adjusted media queries for footer to make it look good on small screens as well
- changed bg-blur of footer so on scrolling it wont show scrolling content going down (In Dark theme )

Closes #255 

## Screenshots

image

image

image

### CODE REVIEW

**Comment by ujjawaltyagii** [comment]
@rupali-codes

**Comment by rupali-codes** [comment]
It's still overflowing when there's even less content. could you fix that? 

image

**Comment by ujjawaltyagii** [comment]
@rupali-codes oh ! On other pages issue was resolved so I missed to check on homepage...my bad. I made the necessary changes and I guess the issue is resolved now.

**Comment by ujjawaltyagii** [comment]
Updated PR !

**Comment by rupali-codes** [comment]
thanks for the changes. but still overflowing other components

image

**Comment by ujjawaltyagii** [comment]
@rupali-codes any idea why it's happening ???  Because on my system (in chrome and opera) it's working fine
 
adding pic :

image

**Comment by rupali-codes** [comment]
@ujjawaltyagii i'm having in both chrome and brave.

**Comment by rupali-codes** [comment]
hey @ujjawaltyagii could you please resolve conflicts?

File: components/Footer/Footer.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ interface FooterProps {}
 
 export const Footer: FC<FooterProps> = ({}) => {
   return (
-    <div className="w-full flex items-baseline justify-center z-10 gap-1 rounded-lg bg-[#ffffff0a] backdrop-blur-md sm:py-3 py-2 px-1 md:mt-4">
+    <div className="bottom-4 dark:bg-[#101623] bg-gray-100 transition-all ease-in duration-300 w-full flex items-baseline justify-center z-10 gap-1 rounded-lg bg-[#ffffff0a] backdrop-blur-md sm:py-3 py-2 px-1 md:mt-[1.9rem]">
```
Comment (CBID2) [inline components/Footer/Footer.tsx:10]:
```suggestion
    <article className="bottom-4 dark:bg-[#101623] bg-gray-100 transition-all ease-in duration-300 w-full flex items-baseline justify-center z-10 gap-1 rounded-lg bg-[#ffffff0a] backdrop-blur-md sm:py-3 py-2 px-1 md:mt-[1.9rem]">
```

File: layouts/GeneralLayout.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ const GeneralLayout = ({ children }: { children: ReactNode }) => {
   return (
     <section>
       <SideNavbar />
-      <div className="mx-auto mt-20 min-h-[calc(100vh-85px)] bg-gray-100 px-4 pb-[1rem] transition-all duration-300 ease-in dark:bg-[#101623] lg:mt-0 lg:pl-[290px]">
+      <div className="mx-auto mt-[4rem] min-h-[calc(100vh-85px)] bg-gray-100 px-4 pb-[1rem] transition-all duration-300 ease-in dark:bg-[#101623] lg:mt-0 lg:pl-[290px]">
```
Comment (CBID2) [inline layouts/GeneralLayout.tsx:9]:
```suggestion
      <article className="mx-auto mt-[4rem] min-h-[calc(100vh-85px)] bg-gray-100 px-4 pb-[1rem] transition-all duration-300 ease-in dark:bg-[#101623] lg:mt-0 lg:pl-[290px]">
```

File: layouts/GeneralLayout.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ const GeneralLayout = ({ children }: { children: ReactNode }) => {
...
         {children}
         <Footer />
       </div>
```
Comment (CBID2) [inline layouts/GeneralLayout.tsx:12]:
```suggestion
      </article>
```

File: pages/index.tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ export default function Home() {
         <meta name="viewport" content="width=device-width, initial-scale=1" />
         <link rel="icon" href="/icon.png" className="rounded-full" />
       </Head>
-      <main className="lg:h-[calc(100vh-85px)] h-[calc(100vh-150px)] flex flex-col">
+      <main className="lg:h-[calc(100vh-99px)] h-[calc(100vh-150px)] flex flex-col">
         <TopBar header="Home" />
         <div className=" m-auto flex flex-col  items-start gap-2">
             <div className="m-auto md:text-7xl text-5xl gap-2 flex items-center justify-center">
```
Comment (CBID2) [inline pages/index.tsx:26]:
```suggestion
        <article className=" m-auto flex flex-col  items-start gap-2">
            <div className="m-auto md:text-7xl text-5xl gap-2 flex items-center justify-center">
```

File: components/cards/index.tsx [hunk 5]
```diff
@@ -13,7 +13,8 @@ const Cards = () => {
   };
   return (
     <div
```
Comment (CBID2) [inline components/cards/index.tsx:15]:
```suggestion
    <article 
```

File: layouts/GeneralLayout.tsx (hunk 2 above)
Comment (CBID2) [inline layouts/GeneralLayout.tsx:9]:
```suggestion
      <article className="mx-auto mt-[4rem] min-h-[calc(100vh-85px)] bg-gray-100 px-4 pb-[1rem] transition-all duration-300 ease-in dark:bg-[#101623] lg:mt-0 lg:pl-[290px]">
```

File: layouts/GeneralLayout.tsx (hunk 3 above)
Comment (CBID2) [inline layouts/GeneralLayout.tsx:12]:
```suggestion
      </article>
```

File: pages/index.tsx (hunk 4 above)
Comment (CBID2) [inline pages/index.tsx:26]:
```suggestion
        <article className=" m-auto flex flex-col  items-start gap-2">
            <article className="m-auto md:text-7xl text-5xl gap-2 flex items-center justify-center">
```

Comment (CBID2) [inline pages/index.tsx:26]:
```suggestion
        <article className=" m-auto flex flex-col  items-start gap-2">
            <article className="m-auto md:text-7xl text-5xl gap-2 flex items-center justify-center">
```

File: components/Footer/Footer.tsx (hunk 1 above)
Comment (CBID2) [inline components/Footer/Footer.tsx:10]:
```suggestion
    <article className="bottom-4 dark:bg-[#101623] bg-gray-100 transition-all ease-in duration-300 w-full flex items-baseline justify-center z-10 gap-1 rounded-lg bg-[#ffffff0a] backdrop-blur-md sm:py-3 py-2 px-1 md:mt-[1.9rem]">
```

File: components/cards/index.tsx (hunk 5 above)
Comment (CBID2) [inline components/cards/index.tsx:15]:
```suggestion
    <article
```

File: layouts/GeneralLayout.tsx (hunk 2 above)
Comment (CBID2) [inline layouts/GeneralLayout.tsx:9]:
```suggestion
      <article className="mx-auto mt-[4rem] min-h-[calc(100vh-85px)] bg-gray-100 px-4 pb-[1rem] transition-all duration-300 ease-in dark:bg-[#101623] lg:mt-0 lg:pl-[290px]">
```

File: layouts/GeneralLayout.tsx (hunk 3 above)
Comment (CBID2) [inline layouts/GeneralLayout.tsx:12]:
```suggestion
      </article>
```

File: pages/index.tsx (hunk 4 above)
Comment (CBID2) [inline pages/index.tsx:26]:
```suggestion
        <article className=" m-auto flex flex-col  items-start gap-2">
            <article className="m-auto md:text-7xl text-5xl gap-2 flex items-center justify-center">
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @ujjawaltyagii! 😄 I made some suggestions on how to improve your pull request. In the future, I highly recommend using more semantic HTML elements instead of `<div>`. They'll make your code more accessible. Other than that, great work! 😄 👍🏾

**Comment by rupali-codes** [comment]
This PR have conflicts that must be resolved :)

**Comment by rupali-codes** [comment]
@ujjawaltyagii Are you still working on this?

**Comment by rupali-codes** [comment]
@ujjawaltyagii are you still working on it?

**Comment by ujjawaltyagii** [comment]
@rupali-codes sorry for being inactive on this PR, actually i was expecting someone to review the responsiveness as earlier things were working fine on my side but there were some issues of overflowing on your side. But as far as I remember this issue has already been resolved in a different PR because I actually reviewed the code of that PR. 
So, mind if I close this PR now ? because I don't think we need these changes anymore (correct me if I'm wrong)

**Pull request closed without merging**